fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/OverworldBiomes.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.biomes.v1;
+
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.BIOME_WEIGHT_LOOKUP;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.CUSTOM_BIOMES;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.EDGE_MAP;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.HILLS_MAP;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.INJECTED_BIOME_LIST;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.RIVER_MAP;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.SHORE_MAP;
+import static net.fabricmc.fabric.impl.biomes.BiomeLists.VARIANTS_MAP;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fabricmc.fabric.api.biomes.v1.RiverAssociates.RiverAssociate;
+import net.fabricmc.fabric.impl.biomes.BiomeAssociate;
+import net.fabricmc.fabric.impl.biomes.VariantAssociate;
+import net.minecraft.util.Pair;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.biome.Biome;
+
+public final class OverworldBiomes
+{
+	private OverworldBiomes() {}
+	
+	/**
+	 * Adds the biome to the specified climate group, with the specified weight
+	 * 
+	 * @param biome the biome to be added
+	 * @param climate the climate group whereto the biome is added
+	 * @param weight the weight of the entry
```

Comment:
Weight needs more explanation.  Give mod authors enough info to come up with useful values without having to read and understand all the downstream code. That makes the API a useful abstraction.

Reply:
So I can't assume all mod makers know probability? lol

Reply:
You can assume mod authors understand the biome will be selected based on the biome weight relative to total weight of all biomes using a uniform probability distribution.  What isn't clear are:

1) What values apply for vanilla biomes, if they have weights? And if vanilla biomes don't have weights, how does this mechanism interact with or alter vanilla biome selection?

2) What values are typical or recommended?  If you offer no guidance at all one author might choose 10 while another chooses 1000 for no particular reason.  You've given them absolutely no basis for selecting reasonable weight values when multiple mods inject biomes.

Reply:
Weight has been clarified in the OverworldClimate enum